### PR TITLE
Supporting mixed-precision (Spec/Tree/Normal) Incremental MultiHead Attention

### PR DESCRIPTION
**Pull request by jiazhihao**
**Description of changes:**

This PR implements mixed-precision support for the three incremental MHA operators

- [x] Single-precision alignment
- [x] Half-precision alignment

**Related Issues:**

Linked Issues:
- Issue #

Issues closed by this PR:
- Closes #

**Before merging:**

- [ ] Did you update the [flexflow-third-party](https://github.com/flexflow/flexflow-third-party) repo, if modifying any of the Cmake files, the build configs, or the submodules?

### CODE REVIEW

**Comment by jiazhihao** [comment]
@xinhaoc @gabrieleoliaro I have finished the initial implementation to support mixed precision in the three IncMHA operators. Both spec_infer and incr_decoding examples work for the example prompts, I haven't tested the alignment thoroughly. It would be great if you can help with the following todos:

- [ ] A more comprehensive alignment test for the three operators
- [ ] Changing the weights to half precision and aligning the output with the single-precision result. We may not be able to fully align the two versions due to floating points but the outputs shouldn't be very different.

**Comment by xinhaoc** [comment]
got it.

**Comment by gabrieleoliaro** [comment]
Working on the alignment test!

File: tests/align/align_utils.py
Comment (xinhaoc) [inline tests/align/align_utils.py:1]: I changed the threshold to pass the conv2d alignment test.

**Pull request merged**